e2e/test/scenarios/dashboard/tabs.cy.spec.js

```diff
@@ -78,6 +78,72 @@ describe("scenarios > dashboard > tabs", () => {
       cy.findByText("Our analytics").should("be.visible");
     });
   });
+
+  it("should only fetch cards on the current tab", () => {
+    visitDashboardAndCreateTab({ dashboardId: 1, save: false });
+
+    // Add card to second tab
+    cy.icon("pencil").click();
+    openQuestionsSidebar();
+    sidebar().within(() => {
+      cy.findByText("Orders, Count").click();
+    });
+    saveDashboard();
+
+    cy.intercept(
+      "POST",
+      `/api/dashboard/1/dashcard/1/card/1/query`,
+      cy.spy().as("firstTabQuery"),
+    );
+    cy.intercept(
+      "POST",
+      `/api/dashboard/1/dashcard/2/card/2/query`,
+      cy.spy().as("secondTabQuery"),
+    );
+
+    // Visit first tab and confirm only first card was queried
+    visitDashboard(1, { params: { tab: 1 } });
+    cy.get("@firstTabQuery").should("have.been.calledOnce");
+    cy.get("@secondTabQuery").should("not.have.been.called");
+
+    // Visit second tab and confirm only second card was queried
+    cy.findByRole("tab", { name: "Tab 2" }).click();
+    cy.get("@firstTabQuery").should("have.been.calledOnce");
+    cy.get("@secondTabQuery").should("have.been.calledOnce");
+
+    // Go back to first tab, expect no additional queries
+    cy.findByRole("tab", { name: "Tab 1" }).click();
+    cy.get("@firstTabQuery").should("have.been.calledOnce");
+    cy.get("@secondTabQuery").should("have.been.calledOnce");
+
+    // Go to public dashboard
+    cy.request("PUT", "/api/setting/enable-public-sharing", { value: true });
+    cy.request("POST", `/api/dashboard/1/public_link`).then(
+      ({ body: { uuid } }) => {
+        cy.intercept(
+          "GET",
+          `/api/public/dashboard/${uuid}/dashcard/1/card/1?parameters=%5B%5D`,
+          cy.spy().as("publicFirstTabQuery"),
+        );
+        cy.intercept(
+          "GET",
+          `/api/public/dashboard/${uuid}/dashcard/2/card/2?parameters=%5B%5D`,
+          cy.spy().as("publicSecondTabQuery"),
+        );
+
+        cy.visit(`public/dashboard/${uuid}`);
```

Comment:
It would be nice to logout before testing, I remember at least one case when public dashboards code did not work for unauthorized users while being perfectly fine for authenticated users

Reply:
Yeah that's a valid concern. I think it may be redundant to test here though since in [`public-dashboard.cy.spec.js`](https://github.com/metabase/metabase/blob/4a2b5dc498b511554823e83d49d7619e1aa28864/e2e/test/scenarios/sharing/public-dashboard.cy.spec.js#L50) we already check if the public link works for all types of users (admin, non-admin, signed out).